File: _240_search_a_2D_matrix_II.java
public class _240_search_a_2D_matrix_II{
    public boolean searchMatrix(int[][] matrix, int target) {
        if(matrix == null){
            return false;
        }
        int rows = matrix.length;
        int columns = matrix[0].length;
        int i = 0, j = columns - 1;
        boolean found = false;

        while(i < rows && j >= 0){
            if (matrix[i][j] == target) {
                found = true;
                break;
            }else if (matrix[i][j] > target) {
                j--;
            }else{
                i++;
            }
        }
        return found;
    }

    public static void main(String[] args){
        _240_search_a_2D_matrix_II a = new _240_search_a_2D_matrix_II();
        int[][] matrix = {{1, 3, 5, 7}, {10, 11, 16, 20}, {23, 30, 34, 50}};
        System.out.println("res: " + a.searchMatrix(matrix, 21));
    }
}

File: _240_search_a_2D_matrix_II.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int rows = matrix.size();
        int columns = matrix[0].size();
        int i = 0, j = columns - 1;
        bool found = false;

        while(i < rows && j >= 0){
            if(matrix[i][j] == target){
                found = true;    
                break;
            }else if(matrix[i][j] > target){
                j--;
            }else{
                i++;
            }
        }

        return found;
    }


};

int main(int argc, char *argv[]){
   Solution s;
   vector<vector<int>> matrix = {{1,4,7,11,15},
                                 {2,5,8,12,19},
                                 {3,6,9,16,22},
                                 {10,13,14,17,24},
                                 {18,21,23,26,30}};
   
   for(auto &row : matrix){
        for(auto col : row){
            cout << col << ",";
        }
        cout << endl;
   }

   cout << "res: " << s.searchMatrix(matrix, 5) << endl;

}
